File: android_kimi/domain/src/main/java/com/transcripto/domain/model/CipherMethod.kt
package com.transcripto.domain.model

/**
 * Represents the available cipher methods for encryption/decryption.
 */
enum class CipherMethod {
    CAESAR,
    BASE64,
    VIGENERE,
    XOR
}

/**
 * Represents the operation mode (encrypt or decrypt).
 */
enum class CipherMode {
    ENCRYPT,
    DECRYPT
}

/**
 * Data class representing encryption/decryption parameters.
 */
data class CipherParams(
    val method: CipherMethod,
    val key: String? = null,
    val shift: Int? = null,
    val salt: String? = null,
    val useSalt: Boolean = false
)

/**
 * Result wrapper for encryption/decryption operations.
 */
sealed class CipherResult {
    data class Success(val data: String) : CipherResult()
    data class Error(val message: String) : CipherResult()
}

/**
 * Parsed envelope data structure.
 */
data class EnvelopeData(
    val method: CipherMethod,
    val salt: String? = null,
    val payload: String
)

File: android_flutter/transcripto/packages/domain/lib/src/cipher_method.dart
abstract class CipherMethod {
  String encrypt(String input, CipherParams params);
  String decrypt(String input, CipherParams params);
}

enum CipherKind { base64, caesar, vigenere, xor }

enum CipherMode { encrypt, decrypt }

class CipherParams {
  final CipherKind method;
  final CipherMode mode;
  final String? key; // for vigenere/xor
  final int? shift; // for caesar
  final List<int>? salt; // optional
  final bool useSalt;
  const CipherParams({
    required this.method,
    required this.mode,
    this.key,
    this.shift,
    this.salt,
    this.useSalt = false,
  });

  CipherParams copyWith({
    CipherKind? method,
    CipherMode? mode,
    String? key,
    int? shift,
    List<int>? salt,
    bool? useSalt,
  }) => CipherParams(
        method: method ?? this.method,
        mode: mode ?? this.mode,
        key: key ?? this.key,
        shift: shift ?? this.shift,
        salt: salt ?? this.salt,
        useSalt: useSalt ?? this.useSalt,
      );
}
